### PR TITLE
Update example component and CLI js

### PR DESCRIPTION
**Pull request by dcwarwick**
Contributes to #393 

Brings the Javascript files for the ExampleComponent and the CLI templates in line with our code guidelines.

#### What did you change?

ExampleComponent js, template js, and CODE_GUIDELINES.md

#### How did you test and verify your work?

Ran a build and ran storybook.

### CODE REVIEW

File: packages/experimental/src/components/ExampleComponent/ExampleComponent.js [hunk 1]
```diff
@@ -53,83 +66,114 @@ export const ExampleComponent = !pkg.isComponentEnabled(componentName)
...
+            /* Pass through any other property values as HTML attributes. */
+            ...rest
+          }>
           <Button
+            className={`${blockClass}__secondary-button`}
             kind={secondaryKind}
             onClick={handleSecondaryClick}
             size={size}
-            disabled={props.disabled}>
+            disabled={rest.disabled}>
```
Comment (davidmenendez) [inline packages/experimental/src/components/ExampleComponent/ExampleComponent.js:91]: shouldn't `disabled` just be defined at the top with the other props?

**Review by davidmenendez — CHANGES_REQUESTED** [review]
looks good. just one comment.

File: packages/experimental/src/components/ExampleComponent/ExampleComponent.js (hunk 1 above)
Comment (dcwarwick) [reply packages/experimental/src/components/ExampleComponent/ExampleComponent.js:91]: Yes, I think so. It looked a bit odd to me too, this. Not sure why it was done. Fixed.

**Comment by matthewgallo** [comment]
Can we add the generated component to the list of default components in `package-settings.js` when the CLI script runs? I think that would make the developer experience of using that script a little better.

**Comment by dcwarwick** [comment]
> Can we add the generated component to the list of default components in `package-settings.js` when the CLI script runs? I think that would make the developer experience of using that script a little better.

That's a nice idea. Needs investigating. It's come up once before, so I've opened #460 to address it.

**Review by lee-chase — APPROVED** [review]

**Review by davidmenendez — APPROVED** [review]

**Pull request merged**